app/scripts/modules/google/serverGroup/configure/serverGroupCommandBuilder.service.js

```diff
@@ -176,18 +176,40 @@ module.exports = angular.module('spinnaker.gce.serverGroupCommandBuilder.service
     function populateCustomMetadata(metadataItems, command) {
       // Hide metadata items in the wizard.
       if (metadataItems) {
+        let customUserData = '';
+        let customUserDataKeys = [];
         if (angular.isArray(metadataItems)) {
           metadataItems.forEach(function (metadataItem) {
-            if (!_.includes(gceServerGroupHiddenMetadataKeys, metadataItem.key)) {
+            if (metadataItem.key === 'customUserData') {
+              customUserData = metadataItem.value;
+              customUserDataKeys = getCustomUserDataKeys(customUserData);
+              command.userData = customUserData;
+            }
+            if (!_.includes(customUserDataKeys, metadataItem.key) && !_.includes(gceServerGroupHiddenMetadataKeys, metadataItem.key)) {
               command.instanceMetadata[metadataItem.key] = metadataItem.value;
             }
           });
         } else {
+          if (metadataItems['customUserData']) {
```

Comment:
Use dot notation for object traversal (e.g., `metadataItems.customUserData`) unless there's a good reason not to do so.